pilot/pkg/xds/ads.go

```diff
@@ -463,9 +464,10 @@ func (s *DiscoveryServer) handleEds(con *Connection, discReq *discovery.Discover
 		return nil
 	}
 
+	changed := listDifference(clusters, con.Clusters)
```

Comment:
This seems fine in ping pong mode, where at most one request/response is on the wire, and no NACK is expected.

Our general rule should be: if any cluster is not explicitly acked by envoy, pilot should include that cluster as well.

Otherwise, it's possible that pilot may assume a cluster A is pushed to envoy, but somehow envoy doesn't apply the cluster A. 
The reasons include but not limited to 
1. stale and skipped response, since the response message order at transport layer is not always reflecting this `handleEds`order 
2. explicit NACK on the wire

Reply:
I remember once I opened the debug log of pilot and found that eds request is not always sent following each cds response.